beamer/deploy/util.py

```diff
@@ -85,22 +85,26 @@ def deploy_contract(web3: Web3, constructor_spec: Union[str, Sequence]) -> Deplo
     return deployed
 
 
-def _resolve_constructor_args(resolver: Contract, constructor_spec: str | Sequence) -> tuple:
+def _resolve_constructor_args(
+    resolver: Contract, request_manager: Contract, constructor_spec: str | Sequence
+) -> tuple:
     if isinstance(constructor_spec, str):
         # This is just the contract's name.
         return (constructor_spec,)
-    return tuple(resolver.address if arg == "${resolver}" else arg for arg in constructor_spec)
+    args: list[str] = []
+    for arg in constructor_spec:
+        if arg == "${resolver}":
+            args.append(resolver.address)
+        elif arg == "${request_manager}":
+            args.append(request_manager.address)
+        else:
+            args.append(arg)
+    return tuple(args)
```

Comment:
I think we should rather handle this in a generic manner at this point, something like

- instead of the `resolver` and the `request_manager` arguments, this function should take a dictionary `dict[str, Contract]`
- have code that does the `${...}` substitution by looking the key in the provided dict